sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/randomExpressions.scala

```diff
@@ -298,8 +281,8 @@ case class Uniform(min: Expression, max: Expression, seedExpression: Expression,
         cast(min, DoubleType),
         Multiply(
           Subtract(
-            cast(max, DoubleType),
-            cast(min, DoubleType)),
+            cast(If(IsNull(max), Literal(null, max.dataType), max), DoubleType),
```

Comment:
hmm is this necessary? Cast is null-propagating, so null input results to null output already.